pkg/netceptor/netceptor.go

```diff
@@ -258,10 +259,11 @@ func (s *Netceptor) Status() Status {
 	}
 	s.serviceAdsLock.RUnlock()
 	return Status{
-		NodeID:         s.nodeID,
-		Connections:    conns,
-		RoutingTable:   routes,
-		Advertisements: serviceAds,
+		NodeID:               s.nodeID,
+		Connections:          conns,
+		RoutingTable:         routes,
+		Advertisements:       serviceAds,
+		KnownConnectionCosts: s.knownConnectionCosts,
```

Comment:
You should be holding a read lock on `s.knownNodeLock` before accessing this.

Reply:
yeah im realizing this is probably all sorts of wrong, i'll fix it shortly